### ISSUE TITLE
Pipewire stream is shared between clients even when they have different mouse pointer preference

### ISSUE THREAD

**Issue by columbarius** [emersion/xdg-desktop-portal-wlr#160]
Start screencast with obs without mouse pointer and then start xdp-screencast-portal.py and it will connect to the same pipewirenode. Now the mouse pointer is visible in both streams.

**Comment by columbarius** [comment]
Partly resolved by #161 

The open issue is, that both screencasts will show a mouse cursor until the one with cursor is closed. Then the cast without cursor will resume to it's cursorless state.

**Comment by ammen99** [comment]
Isn't this a general issue with screencopy/export-dmabuf (or maybe even the wlroots impl)?

**Comment by emersion** [comment]
The wlroots protocols allow specifying whether the cursor should be included in the screencast. The issue is with PipeWire streams shared between multiple consumers.

**Comment by columbarius** [comment]
Not in this case. After #161 I can see two different streams in `pw-dot --detail && xdot pw.dot`

**Comment by emersion** [comment]
Oh, interesting. Then indeed this is a general issue with the screencopy/export-dmabuf protocol. Enabling the cursor in screencopy works by disabling the hw cursor.

**Comment by columbarius** [comment]
So the cursor gets rendered onto the output buffer. And screencopy probably just copies that?

I also had performance issues when using dmabuf and shm screencast on the same output, but that can have other reasons atm.

**Comment by emersion** [comment]
Yeah, screencopy just copies the whole final buffer. When one screencopy client asks for the cursor image, the image gets painted to the final buffer, and all screencopy clients get it.

**Comment by columbarius** [comment]
It probably makes more sense to work on implementing #133 instead of trying to copy one frame before rendering the cursor and one after?

**Comment by emersion** [comment]
Unsure… #133 would require a lot of protocol work, "just" for a small feature. OTOH, not sure how to hook screencopy into the compositor rendering process either.

**Comment by columbarius** [comment]
Didn't say it would be easier, but we could keep the hardware cursor, do proper/optimized damage reporting and the client can decide if it want's to show the cursor or not.
